include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -0,0 +1,347 @@
+                         std::back_inserter(sequence));
+        }
+        else
+        {
+            detail::consume(stream_view | view::take_until(is_end));
+        }
+        //Jump over //
+        detail::consume(stream_view | view::take_until(is_cntrl));
+        ++stream_it;
+
+
+        // make sure "buffer at end" implies "stream at end"
+        if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+            (!stream.eof()))
+        {
+            stream.get(); // triggers error in stream and sets eof
+        }
+    }
+
+    //!\copydoc sequence_file_output_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void write(stream_type                     & stream,
+               sequence_file_output_options const & options,
+               seq_type                       && sequence,
+               id_type                        && id,
+               qual_type                      && SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+
+        ranges::ostreambuf_iterator stream_it{stream};
+        size_t sequence_size = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size = ranges::size(sequence);
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing embl files."};
+        }
+        else
+        {
+            if (options.complete_header)
+            {
+                ranges::copy(id, stream_it);
+            }
+            else
+            {
+                if (ranges::empty(id)) //[[unlikely]]
+                    throw std::runtime_error{"The ID field may not be empty when writing embl files."};
+
+                stream_it = 'I';
+                stream_it = 'D';
+                stream_it = ' ';
+                ranges::copy(id, stream_it);
+                stream_it = ';';
+                stream_it = ' ';
+                ranges::copy(std::to_string(sequence_size), stream_it);
+                ranges::copy(std::string_view{" BP.\nXX\n"}, stream_it);
+            }
+
+        }
+
+        // Sequence
+        if constexpr (detail::decays_to_ignore_v<seq_type>) // sequence
+        {
+            throw std::logic_error{"The SEQ and SEQ_QUAL fields may not both be set to ignore when writing embl files."};
+        }
+        else
+        {
+            if (ranges::empty(sequence)) //[[unlikely]]
+                throw std::runtime_error{"The SEQ field may not be empty when writing embl files."};
+
+            ranges::copy(std::string_view{"SQ Sequence "}, stream_it);
+            ranges::copy(std::to_string(sequence_size), stream_it);
```

Comment:
Do we really need the sequence size twice – in ID and again in SQ?

Reply:
No, but there is no harm to have it at both positions and in most examples it is given twice, e.g. here: https://www.genomatix.de/online_help/help/sequence_formats.html